extensions/notebook/src/dialog/managePackages/installedPackagesTab.ts

```diff
@@ -0,0 +1,160 @@
+import * as azdata from 'azdata';
+
+import JupyterServerInstallation, { PythonPkgDetails } from '../../jupyter/jupyterServerInstallation';
+import * as utils from '../../common/utils';
+import { ManagePackagesDialog } from './managePackagesDialog';
+import CodeAdapter from '../../prompts/adapter';
+import { QuestionTypes, IQuestion } from '../../prompts/question';
+
+const localize = nls.loadMessageBundle();
+
+export class InstalledPackagesTab {
+	private prompter: CodeAdapter;
+
+	private installedPkgTab: azdata.window.DialogTab;
+
+	private installedPackageCount: azdata.TextComponent;
+	private installedPackagesTable: azdata.TableComponent;
+	private installedPackagesLoader: azdata.LoadingComponent;
+	private uninstallPackageButton: azdata.ButtonComponent;
+
+	private readonly UninstallButtonText = localize('managePackages.uninstallButtonText', "Uninstall selected packages");
+	private readonly InstalledTabTitle = localize('managePackages.installedTabTitle', "Installed");
+	private readonly PkgNameColumn = localize('managePackages.pkgNameColumn', "Name");
+	private readonly PkgVersionColumn = localize('managePackages.newPkgVersionColumn', "Version");
+
+	constructor(private dialog: ManagePackagesDialog, private jupyterInstallation: JupyterServerInstallation) {
+		this.prompter = new CodeAdapter();
+
+		this.installedPkgTab = azdata.window.createTab(this.InstalledTabTitle);
+
+		this.installedPkgTab.registerContent(async view => {
+			this.installedPackageCount = view.modelBuilder.text().withProperties({
+				value: ''
+			}).component();
+
+			this.installedPackagesTable = view.modelBuilder.table()
+				.withProperties({
+					columns: [this.PkgNameColumn, this.PkgVersionColumn],
+					data: [[]],
+					height: '600px',
+					width: '400px'
+				}).component();
+
+			this.uninstallPackageButton = view.modelBuilder.button()
+				.withProperties({
+					label: this.UninstallButtonText,
+					width: '200px'
+				}).component();
+			this.uninstallPackageButton.onDidClick(() => this.doUninstallPackage());
+
+			let formModel = view.modelBuilder.formContainer()
+				.withFormItems([{
+					component: this.installedPackageCount,
+					title: ''
+				}, {
+					component: this.installedPackagesTable,
+					title: ''
+				}, {
+					component: this.uninstallPackageButton,
+					title: ''
+				}]).component();
+
+			this.installedPackagesLoader = view.modelBuilder.loadingComponent()
+				.withItem(formModel)
+				.withProperties({
+					loading: true
+				}).component();
+
+			await view.initializeModel(this.installedPackagesLoader);
+
+			await this.loadInstalledPackagesInfo();
+		});
+	}
+
+	public get tab(): azdata.window.DialogTab {
+		return this.installedPkgTab;
+	}
+
+	public async loadInstalledPackagesInfo(): Promise<void> {
+		let packageData: string[][];
+		await this.installedPackagesLoader.updateProperties({ loading: true });
+		try {
+			let pythonPackages = await this.jupyterInstallation.getInstalledPipPackages();
```

Comment:
Can this ever appear if the install isn't complete? Do we need to handle that case? I'm likely just being paranoid 😄

Reply:
Currently the Manage Packages button appears as enabled during install. I've opened a separate issue for it here: https://github.com/microsoft/azuredatastudio/issues/5955